<launch>
  <arg name="map_path" default="$(find-pkg-share conveyor_utils)/launch/lab.yaml"/>

  <arg name="belt_speed" default="0.1"/>
  <arg name="package_length" default="0.5"/>
  <arg name="n_packs" default="9"/>
  <arg name="n_reds" default="3"/>

  <arg name="n_runs" default="5"/>
  <arg name="update_rate" default="10."/>
  <arg name="sim_imu" default="False"/>
  <arg name="do_relloc" default="False"/>

  <arg name="led_on_period" default="3."/>
  <arg name="relloc_leds" default="['led_1', 'led_2']"/>

  <arg name="log" default="False"/>
  <arg name="log_rate" default="10"/>
  <arg name="exp_name" default=""/>
  <arg name="log_path" default="/home/gabri/Documents/logs_iros_test/"/>
  <arg name="interface" default="PUI"/>

  <arg name="publish_rate" default="50."/>
  <arg name="gt_x" default="0"/>
  <arg name="gt_y" default="0"/>
  <arg name="gt_yaw" default="0"/>
  <arg name="rotation_topic" default="/metawear_ros/rotation"/>
  <arg name="user_kinematics" default="human"/>


  <arg name="time_for_selection" default=".5"/>
  <arg name="pointer_size" default=".2"/>
  <arg name="pointing_tol" default=".7"/>
  <arg name="pointing_cmap" default=""/>

  <arg name="joy_dev" default="0"/>


  <node name="pack_generator" pkg="iros" exec="pack_generator" output="screen" args="--ros-args --log-level info">
    <param name="belt_speed" value="$(var belt_speed)"/>
    <param name="package_length" value="$(var package_length)"/>
    <param name="n_packs" value="$(var n_packs)"/>
    <param name="n_reds" value="$(var n_reds)"/>
  </node>

  <node name="supervisor" pkg="iros" exec="experiment" output="screen" args="--ros-args --log-level info">
    <param name="map_path" value="$(var map_path)"/>
    <param name="n_runs" value="$(var n_runs)"/>
    <param name="update_rate" value="$(var update_rate)"/>
    <param name="sim_imu" value="$(var sim_imu)"/>
    <param name="do_relloc" value="$(var do_relloc)"/>
  </node>

  <include file="$(find-pkg-share conveyor_utils)/launch/lab.launch">
    <arg name="map_path" value="$(var map_path)" />

    <arg name="belt_speed" value="$(var belt_speed)" />
    <arg name="package_length" value="$(var package_length)" />
    <arg name="pointing_tol" value="$(var pointing_tol)"/>
    <arg name="pointing_cmap" value="$(var pointing_cmap)"/>
    <arg name="joy_dev" value="$(var joy_dev)" />

  </include>

  <include file="$(find-pkg-share volaly_kinematics)/launch/human_kinematics.launch">
    <arg name="publish_rate" value="$(var publish_rate)" />
    <arg name="rotation_topic" value="$(var rotation_topic)" />
    <arg name="user_kinematics" value="$(var user_kinematics)" />
  </include>

  <node name="task_logger" pkg="iros" exec="task_logger" output="screen" args="--ros-args --log-level info" if="$(var log)">
    <param name="log_rate" value="$(var log_rate)"/>
    <param name="exp_name" value="$(var exp_name)"/>
    <param name="log_path" value="$(var log_path)"/>

    <param name="map_path" value="$(var map_path)"/>
    <param name="belt_speed" value="$(var belt_speed)"/>

    <param name="package_length" value="$(var package_length)"/>
    <param name="n_packs" value="$(var n_packs)"/>
    <param name="n_reds" value="$(var n_reds)"/>
    <param name="n_runs" value="$(var n_runs)"/>
    <param name="interface" value="$(var interface)"/>
    <param name="update_rate" value="$(var update_rate)"/>
  </node>

  <node name="relloc_executor" pkg="relloc" exec="relloc_exec" output="screen" args="--ros-args --log-level info" if="$(var do_relloc)">
    <param name="map_path" value="$(var map_path)"/>
    <param name="led_on_period" value="$(var led_on_period)"/>
    <param name="relloc_leds" value="$(var relloc_leds)"/>
  </node>

  <node name="relloc_node" pkg="relloc" exec="relloc_node" output="screen" args="--ros-args --log-level info">
    <param name="publish_rate" value="$(var publish_rate)" />
    <param name="led_on_period" value="$(var led_on_period)"/>

  </node>

  <node name="pui_node" pkg="conveyor_pui" exec="pui_node" output="screen" args="--ros-args --log-level info">
    <param name="map_path" value="$(var map_path)"/>
    <param name="time_for_selection" value="$(var time_for_selection)"/>
    <param name="pointer_size" value="$(var pointer_size)"/>
    <param name="pointing_tol" value="$(var pointing_tol)"/>
  </node>

  <node name="world_human_gt_bc" pkg="tf2_ros" exec="static_transform_publisher" args="$(var gt_x) $(var gt_y) 0 $(var gt_yaw) 0 0 world human_gt" unless="$(eval '$(var do_relloc) or $(var sim_imu)')"/>
  <node name="human_gt_human_foot_bc" pkg="tf2_ros" exec="static_transform_publisher" args="0 0 0 0 0 0 1 human_gt human_footprint" unless="$(var do_relloc)"/>

</launch>
